portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,20 +14,96 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Comment;
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
 
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private final static String tblTitle = "Comment";
+  private final static String tblComment = "comment";
+  private final static String tblName = "name";
+  private final static String tblTime = "timestamp";
+
+  private final static String htmlComment = "text-input";
+  private final static String htmlName = "name";
+  private final static String htmlLimit = "commentLimit";
+
+  private static int limit = -1;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jacob!</h1>");
+    Query query = new Query(tblTitle).addSort(tblTime, SortDirection.DESCENDING);;
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<Comment>();
+    int count = 0;
+    for (Entity entity : results.asIterable()) {
+      if (limit > 0) {
+        if(count < limit){
+        String commentString = (String) entity.getProperty(tblComment);
+        String name = (String) entity.getProperty(tblName);
+        long id = entity.getKey().getId();
+
+        Comment comment = new Comment(id, commentString, name);
+        comments.add(comment);
+        count++;
+        }else {
+          break;
+        }
+      }else {
+        String commentString = (String) entity.getProperty(tblComment);
+        String name = (String) entity.getProperty(tblName);
+        long id = entity.getKey().getId();
+
+        Comment comment = new Comment(id, commentString, name);
+        comments.add(comment);
+      }
+    }
```

Comment:
This logic needs some cleanup. I can clearly see some duplications.

Reply:
I worked on the logic to avoid the duplicates.